src/util/emitter.ts

```diff
@@ -56,6 +56,17 @@ export class Emitter {
     // remove specific handler
     for (let i = 0; i < callbacks.length; i++) {
       const { fn: cb, scope: context } = callbacks[i]
+
+      // handle unsubscribing from all callbacks for a given record path
+      if (event !== '' && fn === undefined && scope === context) {
+        callbacks.splice(i, 1)
+      }
+
+      // handle unsubscribing from all callbacks for a list
+      if (event === '' && fn === undefined && scope === context && scope.constructor.name === 'List') {
```

Comment:
whats the difference between a record and list in this context? 

Also an emitter should not know about data types, since it's used for all pubsub client side

Reply:
Yes I'm not very happy with this. 

I digged into it again and found a bug that affects this logic. Will make a new commit to further discuss it